### PR TITLE
Add DISCOVERY_HOST environment variable

### PR DESCRIPTION
**Pull request by radeksimko**
As discussed in #24 

Validation for input added including tests for that validation which should explain how it works and confirm that it works :smiley: 

btw. I'm not sure what was the point of adding the `host` flag and not using it anywhere => I've pointed the `devweb` wrapper to this app, so it can actually use that flag.

### CODE REVIEW

**Comment by radeksimko** [comment]
Is there anything preventing this from merging? Any feedback welcomed @philips

**Comment by yichengq** [comment]
I would prefer to accept separate PRs, each of which targets one specific goal. The PR says to introduce 	DISCOVERY_HOST, but it adds a lot of checking stuffs. Doing this can help us to review this much easily and push forward this more quickly.

I will make limited comment on env-related code.

File: handlers/new.go [hunk 1]
```diff
@@ -9,13 +8,15 @@ import (
...
+	"net/url"
+	"os"
 	"path"
 	"strconv"
 
 	"github.com/coreos/go-etcd/etcd"
 )
 
-var baseURI = flag.String("host", "https://discovery.etcd.io", "base location for computed token URI")
+var baseURI = flag.String("host", os.Getenv("DISCOVERY_HOST"), "base location for computed token URI")
```
Comment (yichengq) [inline handlers/new.go]: this breaks the default value for `host`. Another way to do it is 1. check env existence 2. set it to this flag is this flag has not been set.

**Comment by tschuy** [comment]
I've added a similar env variable, `DISC_HOST`, with the rest of the config added in #43.

**Pull request closed without merging**